### ISSUE TITLE
Add support for Mac OS

### ISSUE THREAD

**Issue by programmingkidx** [gphalkes/tilde#109]
Please add support for building Tilde on Mac OS. I think this awesome software would benefit a lot of us Mac users.

**Comment by n7ihq** [comment]
Tilde would be a great replacement for the nano editor on MacOS.

**Comment by herbygillot** [comment]
Tilde has been added to MacPorts - https://ports.macports.org/port/tilde/

**Comment by programmingkidx** [comment]
👍

**Comment by n7ihq** [comment]
That’s great! I will check it out.

Thanks,

Jim

**Comment by spinicist** [comment]
This is the first time I have been sad I'm a homebrew user!

**Comment by spinicist** [comment]
Okay - I have swapped from Homebrew to Macports specifically to get Tilde. It has installed fine, but an annoyance is that the Command key is not interpreted as Meta. I am using the Esc workaround for now. Am I wrong to expect Cmd to act as Meta?

**Comment by programmingkidx** [comment]
@spinicist Not at all. It is what we all are use to. I would ask the author for support changing this but it looks like he no longer works on anything here on Github. 

@gphalkes Are you still supporting Tilde?

**Comment by gphalkes** [comment]
> @gphalkes Are you still supporting Tilde?

I unfortunately don't really have time to do so, although I would like
to :.-( If there are patches outstanding, I can have a look at
integrating them, which shouldn't take much time.

**Comment by herbygillot** [comment]
> @gphalkes Are you still supporting Tilde?
> I unfortunately don't really have time to do so, although I would like to :.-( If there are patches outstanding, I can have a look at integrating them, which shouldn't take much time.

@gphalkes, can you look at https://github.com/aligrudi/neatroff/pull/6

**Comment by programmingkidx** [comment]
Currently tilde's copy command does not copy to the Mac OS clipboard. I made a fix for this. The dependency libt3widget is where the changes take place. 
What I did first was build tilde from source by following the directions here: https://trac.macports.org/ticket/68594.
Download this archive and extract the files from it: [files.zip](https://github.com/gphalkes/tilde/files/14643800/files.zip).
Then I open the libt3widget folder.
Then:
- replace the file makefile with the one from the archive.
- copy the files mac_clipboard.h and mac_clipboard.m to the src/widgets/ folder.
- replace the file src/widgets/editwindow.cc with the editwindow.cc from the archive.
- Run this command: LIBTOOL=/opt/local/bin/glibtool CXXFLAGS=-I/opt/local/include/ LDFLAGS=-L/opt/local/lib/ LDLIBS=-lunistring CFLAGS=-I/opt/local/include/ ./configure --without-gettext 
- Run this command: make -j 4 all && sudo make install _pkgconfigdir=$PKG_CONFIG_PATH
- - Note: the 4 in  "-j 4" represents the number of CPU cores you have.
- Run this command: glibtool --finish /usr/local/lib
- Delete to the tilde folder.
- Unzip the archive tilde-1.1.3.tar.bz2.
- cd into the folder.
- Run this command: LIBTOOL=/opt/local/bin/glibtool CXXFLAGS="-I/opt/local/include/" LDFLAGS=-L/opt/local/lib/ LDLIBS=-lunistring CFLAGS=-I/opt/local/include/ ./configure --without-gettext
- Run this command: make -j 4 all
- You should see an error at the end that states this: 
```
LD] src/tilde
ld: can't open output file for writing: src/tilde, errno=21 for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [src/tilde] Error 1
```
Run this command to delete src/tilde: rm src/tilde
Run this command: make -j 4 all

This should lead to a working tilde. Run ./src/tilde to try copying. Open a file in tilde and copy some of the text. In another text application paste what is in the clipboard. You should see what was copied in tilde. 

@herbygillot Could you try this out and maybe add it to the tilde port in macports if you like it?

**Comment by SecUnit75** [comment]
Absolutely cannot get Tilde to install on MacOS 14 (Sonoma) on an M3Max ARM/Apple Silicon MBP.

Macports installs correctly, reports version, etc.

sudo port install tilde    ... fails with "Error: Port tilde not found"

Trying to re-sync or selfupdate port doesn't help.

Manually trying to build from the github repo also fails at the last hurdle (missing a t3widget.api.h file apparently).

Any chance of a binary for MacOS which "just works"? :-)

**Comment by programmingkidx** [comment]
I am using Mac OS 12.3 with MacPorts version 2.10.7. I was able to install tilde using this command: 
sudo port install tilde

Are you still not able to install tilde? If so which version of MacPorts are you using? 'port -v' will give you the version number.
